python/lsst/pipe/base/_task_metadata.py

```diff
@@ -149,14 +149,14 @@ def to_dict(self) -> Dict[str, Any]:
         used when a simple dictionary is needed.  Use
         `TaskMetadata.from_dict()` to convert it back.
         """
-        d = {}
+        d: Dict[str, Any] = {}
```

Comment:
I assume you mean
```
Union[TaskMetadata, str, StrictFloat, StrictInt, StrictBool, StrictStr, List[StrictFloat], List[StrictInt], List[StrictBool], List[StrictStr]
```?

Can the `str | StrictStr` within that be collapsed to just `str`?

I think I'll try this but bail out if it doesn't just work.

Reply:
Oh, that first `str` is a mistake and yes it was meant to be union of all of them -- copy and paste error. These are all the types defined in the TaskMetadata pydantic model. I'm not sure if `str`, `float` etc work in the type annotations but the model itself needs StrictStr -- pydantic didn't do the right thing if it was not StrictStr -- it would cast the numbers to strings because the order of types matters to it.

Reply:
It turns out this one's tricky because conversion from `TaskMetadata` to `dict` here is recursive, and hence the type annotation would need to be recursive as well in order to be complete.  So I'll just stick to `Any` hre.